client/src/modules/dashboard/Events/components/SendEmailModal.tsx

```diff
@@ -0,0 +1,109 @@
+import { Button } from '@chakra-ui/button';
+import { Checkbox } from '@chakra-ui/checkbox';
+import { Stack } from '@chakra-ui/layout';
+import {
+  Modal,
+  ModalBody,
+  ModalContent,
+  ModalHeader,
+  ModalCloseButton,
+  ModalFooter,
+  ModalOverlay,
+} from '@chakra-ui/modal';
+import { Alert, AlertIcon, AlertDescription } from '@chakra-ui/react';
+import React from 'react';
+import { useForm } from 'react-hook-form';
+import { useSendEventInviteMutation } from 'generated/graphql';
+interface SendEmailModalProps {
+  onClose: () => any;
+  isOpen: boolean;
+  eventId: number;
+}
+const SendEmailModal: React.FC<SendEmailModalProps> = ({
+  onClose,
+  isOpen,
+  eventId,
+}) => {
+  const {
+    register,
+    getValues,
+    handleSubmit,
+    formState: { errors },
+  } = useForm();
+
+  const checkAtLeastOne = () => {
+    return getValues('confirmed') ||
+      getValues('on_waitlist') ||
+      getValues('canceled')
+      ? true
+      : 'Please tell me if this is too hard.';
+  };
+
+  const [publish] = useSendEventInviteMutation();
+  const onSubmit = (data: any) => {
+    console.log(data);
+    const emailGroups = [];
+    for (const key of Object.keys(data)) {
+      if (data[key]) {
+        emailGroups.push(key);
+      }
+    }
+    console.log(emailGroups);
+
+    publish({ variables: { id: eventId, emailGroups: emailGroups } });
+    onClose();
+  };
+  return (
+    <Modal isOpen={isOpen} onClose={onClose}>
+      <ModalOverlay />
+      <ModalContent>
+        <ModalHeader>Send Email to Attendees</ModalHeader>
+        <ModalCloseButton />
+        <ModalBody>
+          <form id="sendemail" onSubmit={handleSubmit(onSubmit)}>
+            <label htmlFor="attendees">
+              Who do you want to send the email to?
+            </label>
+            <Stack spacing={10} direction="row">
+              <Checkbox
+                {...register('confirmed', { validate: checkAtLeastOne })}
+                defaultChecked
```

Comment:
```suggestion
```
If we use `defaultValues` this is no longer needed.

Reply:
Removed defaultChecked.